streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -83,10 +90,67 @@ public void init(final StateStoreContext context,
         streamThread = Thread.currentThread();
     }
 
-    Position getPosition() {
+    public Position getPosition() {
         return position;
```

Comment:
Should we merge in the wrapped store's position for the same reason you merge in the wrapped store's position below?

Reply:
I am not so sure about this. We won't have a way to know at what position the cache is, even just for debugging purposes.

Reply:
Fair enough. Maybe merging the underlying position into the cache's position here would be too much, but it does seem necessary to merge them together as the result of this method. Otherwise, the cache's position will "shadow" the underlying store's, and callers would have no way to learn the total position of the layered store.